jeneamocan/script.rb

```diff
@@ -0,0 +1,204 @@
+  attr_reader :account
+
+  def initialize
+    run
+    account_info
+    store
+  end
+
+  def account_info
+    @account = account_check
+  end
+
+  def account_check
+    authentication_check
+    puts "Fetching account informatrion"
+    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.NEW_CARDS_ACCOUNTS")
+    page = parse
+    hash = {"accounts" => []}
+    i = page.css('div[class="contracts-section"]').map do |i|
+      unless i.css('div[class="section-title no-data-error"]').any?
+        name = i.css('div[class="main-info"]').css('a[class="name"]').text
+        balance = i.css('div[class="primary-balance"]').css('span[class="amount"]').text
+        currency =  i.css('div[class="primary-balance"]').children.last.text
+        nature = i.css('div[class="section-title h-small"]').text.downcase.capitalize
+        res = {name: name, currency: currency, balance: balance, nature: nature}
+        hash["accounts"] << res
+      else
+      end
+    end
+    hash
+  end
+
+  def store
+    Dir.mkdir(DATA_DIR) unless File.exists?(DATA_DIR)
+    file_name = "#{DATA_DIR}/#{File.basename('accounts')}.json"
+    File.open(file_name, 'w'){|file| file.write(JSON.pretty_generate(@account))}
+    puts "Accounts saved to #{file_name}"
+  end
+end
+
+class Transactions < VB_WebBanking
+  attr_reader :transaction, :transactions
+
+  def transaction_info
+    @last_transaction = last_transaction
+    @transactions = transactions_2months
+  end
+
+  def initialize
+    run
+    transaction_info
+    store
+  end
+
+  def last_transaction
+    authentication_check
+    puts "Fetching latest transaction"
+    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.CP_HISTORY")
+    month = browser.div(class: "month-delimiter").text
+    day = browser.div(class: "day-header").text.split[0]
+    time = browser.span(class: "history-item-time").text
+    date = day + " " + month + " " + time
+    description = browser.a(class: "operation-details").text
+    amount = browser.span(class: %w"history-item-amount transaction").text
+    {"transactions" => [{date: date, description: description, amount: amount}]}
+  end
+
+  def transactions_2months
+    authentication_check
+    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.CP_HISTORY")
+    set_date
+    puts "Fetching transactions for the last two months"
+    sleep(2)
+    page = parse
+    hash = {"transactions" => []}
+    page.css('li[class="history-item success "]').map do |i|
+      month = i.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text
+      day = i.parent.parent.css('div[class = "day-header"]').text
+      time = i.css('span[class= "history-item-time"]').text
+      date = day + " " + month + " " + time
+      description = i.css('span[class="history-item-description"]').text.split.join(" ")
+      if not i.css('span[class="history-item-amount transaction income"]').text.empty?
+        amount = i.css('span[class="history-item-amount transaction income"]').text
```

Comment:
instead of `'span[class="history-item-amount transaction income"]'`

update css format  
`'span.history-item-amount.transaction income'`